hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -203,11 +221,10 @@ protected List<HoodieFileGroup> buildFileGroups(Stream<HoodieBaseFile> baseFileS
       if (baseFiles.containsKey(pair)) {
         baseFiles.get(pair).forEach(group::addBaseFile);
       }
-      if (logFiles.containsKey(pair)) {
-        logFiles.get(pair).forEach(group::addLogFile);
-      }
-
       if (addPendingCompactionFileSlice) {
+        // pending compaction file slice must be added before log files so that
+        // the log files completed later than the compaction instant time could be included
+        // in the file slice with that compaction instant time as base instant time.
```

Comment:
how about pending log compaction?

Reply:
they are handled in the same way of master code.